src/instruments/options/finite_difference_pricer.rs

```diff
@@ -0,0 +1,491 @@
+            * phi[last-1]
+        );
+
+        let theta_n = theta.pop().unwrap();
+        phi.pop();
+        phi.reverse();
+
+        let mut value: f64;
+        let mut inverse_matrix: Vec<Vec<f64>> = Vec::new();
+        let mut matrix_row: Vec<f64> = Vec::new();
+
+        for i in 0..tridiagonal_matrix.len() {
+            for j in 0..tridiagonal_matrix.len() {
+                value = (-1.0_f64).powi((i+j) as i32);
+                
+                if i < j {
+                    for k in i..j {
+                        value *= tridiagonal_matrix[k].last().unwrap();
+                    }
+                    value *= theta[i] * phi[j] / theta_n;
+
+                } else if i == j {
+                    value *= theta[i] * phi[i] / theta_n
+                } else {
+                    for k in j..i {
+                        value *= tridiagonal_matrix[k+1].first().unwrap();
+                    }
+                    value *= theta[j] * phi[i] / theta_n
+                }
+                matrix_row.push(value);
+            }
+            
+            inverse_matrix.push(matrix_row.clone());
+            matrix_row.clear()
+        }
+
+        inverse_matrix
+        
+    }
+
+    fn sub_diagonal(&self, scaler: f64) -> Box<dyn Fn(f64) -> f64 + '_> {
+        let function = move |m: f64| scaler * ((self.volatility.powi(2) * m.powi(2)) - (self.risk_free_rate * m));
+        Box::new(function)
+    }
+
+    fn diagonal(&self, scaler: f64) -> Box<dyn Fn(f64) -> f64 + '_> {
+        let function = move |m: f64| 1.0 + scaler * ((self.volatility.powi(2) * m.powi(2)) + self.risk_free_rate);
+        Box::new(function)
+    }
+
+    fn super_diagonal(&self, scaler: f64) -> Box<dyn Fn(f64) -> f64 + '_> {
+        let function = move |m: f64| scaler * ((self.volatility.powi(2)) * m.powi(2) + (self.risk_free_rate * m));
+        Box::new(function)
+    }
+
+    fn price_steps(&self) -> u32 {
+        (self.initial_price / 100.00) as u32 * 100 + 100
+    }
+
+    fn delta_t(&self, time_steps: u32) -> f64 {
+        self.time_to_maturity as f64 / ((365 * time_steps) as f64)
+    }
+
+    fn time_steps(&self) -> u32 {
+        ((self.time_to_maturity / 365) + 1) as u32 * 1000
+    }
+
+    fn payoff(&self, s: f64) -> f64 {
+        match self.type_flag {
+            TypeFlag::Call => (s - self.strike_price).max(0.0),
+            TypeFlag::Put => (self.strike_price - s).max(0.0)
+        }
+    }
+
+    fn american_time_stop_step(&self, u: Vec<f64>, price_steps: u32) -> Vec<f64> {
+        (0..(price_steps - 1)).map(|i: u32| u[i as usize].max(self.payoff((i+1) as f64 * (2.0 * self.initial_price) / (price_steps as f64)))).collect()
+    }
+
+    fn boundary_condition_at_time_n(&self, price_steps: u32) -> Vec<f64> {
+        let u = (1..(price_steps)).map(|i| self.payoff(((i) as f64) * (2.0 * self.initial_price / (price_steps as f64)))).collect();
+        u
+    }
+    
+    fn call_boundary(&self, t: u32, delta_t: f64) -> f64 {
```

Comment:
There is a bit of a dependency on `365` in the logic, however this restricts the pricer to pricing options in a daily setting with the assumption that there are 365 days in a year. Time/date should be handled by the `time` module so it is more flexible.